src/Bot.ts

```diff
@@ -92,14 +93,44 @@ export class Bot extends builder.UniversalBot {
     }
 
     // set incoming event to any because membersAdded is not a field in builder.IEvent
-    private getConversationUpdateHandler(bot: builder.UniversalBot): (event: any) => void {
-        return async function(event: any): Promise<void> {
+    private getConversationUpdateHandler(bot: builder.UniversalBot): (event: builder.IConversationUpdate) => void {
+        return async function(event: builder.IConversationUpdate): Promise<void> {
+            // We are only interested in member add events
+            if (!event.membersAdded || (event.membersAdded.length === 0)) {
+                return;
+            }
+
             let session = await loadSessionAsync(bot, event);
 
-            if (event.membersAdded && event.membersAdded[0].id && event.membersAdded[0].id.endsWith(config.get("bot.botId"))) {
-                session.send(Strings.bot_introduction); // probably only works in Teams
+            // Determine if the bot was added to the conversation
+            let botId = event.address.bot.id;
+            let botAdded = event.membersAdded && event.membersAdded.find(member => (member.id === botId));
+
+            if (!event.address.conversation.isGroup) {
+                // 1:1 conversation event
+                // If the user hasn't received a first-run message YET, send a message to the user,
+                // introducing your bot and what it can do. Do NOT send this blindly, as you can receive
+                // spurious conversationUpdate events, especially if you use proactive messaging.
+
+                if (!session.userData.freSent) {
+                    session.userData.freSent = true;
+                    session.send(Strings.bot_introduction);
+                } else {
+                    // First-run message has already been sent, so skip sending it again
+                    // Do not remove the check for "freSent" above. Your bot can receive spurious conversationUpdate
+                    // activities from chat service, so if you always respond to all of them, you will send random
+                    // welcome messages to users who have already received the welcome.
+                }
             } else {
-                session.send(Strings.bot_welcome_to_new_person);
+                // Team event (bot or user was added to a team)
```

Comment:
Do we want to future proof this statement and say "Not 1:1" or list "Team or Group chat" - my worry is that someone could read this and think "if (address.conversation.isGroup) is valid, robust way to check if the message is from a channel".